### ISSUE TITLE
Mixed ANOVA fails with categorical variables (KeyError for between factor)

### ISSUE THREAD

**Issue by OlafHaag** [raphaelvallat/pingouin#122]
Hi! I just noticed that when any of the _within_, _subject_, or _between_ variables are of type category, the mixed ANOVA fails with KeyError.

Python: 3.8.3
Numpy: 1.19.1
Pandas: 1.1.1
Pingouin: 0.3.7

To reproduce:
```python
import pandas as pd
import pingouin as pg
import numpy as np

# Generate some data.
blocks = np.tile([1,2,3], 20)
users = np.repeat(range(20), 3)
condition = np.tile(np.repeat(['a', 'b', 'c'], 3), 7)[:60]
df = pd.DataFrame({'subject': users, 'block': blocks, 'condition': condition, 'value': np.random.normal(size=60)})

# Make sure it works the first time.
aov_working = pg.mixed_anova(data=df, dv='value', within='block', subject='subject', between='condition')

# You can set any of the within, subject, or between variables to type category.
df[['subject', 'block', 'condition']] = df[['subject', 'block', 'condition']].astype('category')

# Watch it fail with KeyError: 'Column not found: condition'.
aov_failing = pg.mixed_anova(data=df, dv='value', within='block', subject='subject', between='condition')
```

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
 in 
     15 df[['subject', 'block', 'condition']] = df[['subject', 'block', 'condition']].astype('category')
     16 # Watch it fail with KeyError: 'Column not found: condition'.
---> 17 aov_failing = pg.mixed_anova(data=df, dv='value', within='block', subject='subject', between='condition', correction=True)

~/miniconda3/envs/pm3/lib/python3.8/site-packages/pingouin/parametric.py in mixed_anova(data, dv, within, subject, between, correction, effsize)
   1423     # should have a unique range / set of values for each between-subject
   1424     # group e.g. group1= 1 --> 20 and group2 = 21 --> 40.
-> 1425     if not (data.groupby([subject, within])[between].nunique() == 1).all():
   1426         raise ValueError("Subject IDs cannot overlap between groups: each "
   1427                          "group in `%s` must have a unique set of "

~/miniconda3/envs/pm3/lib/python3.8/site-packages/pandas/core/groupby/generic.py in __getitem__(self, key)
   1646                 stacklevel=2,
   1647             )
-> 1648         return super().__getitem__(key)
   1649 
   1650     def _gotitem(self, key, ndim: int, subset=None):

~/miniconda3/envs/pm3/lib/python3.8/site-packages/pandas/core/base.py in __getitem__(self, key)
    226         else:
    227             if key not in self.obj:
--> 228                 raise KeyError(f"Column not found: {key}")
    229             return self._gotitem(key, ndim=1)
    230 

KeyError: 'Column not found: condition'
```

**Comment by raphaelvallat** [comment]
Hi @OlafHaag !

Thanks for opening the issue. You're absolutely right, the issue comes from this specific line:

https://github.com/raphaelvallat/pingouin/blob/08560bb978949e97653b203f7374e90b022b88fb/pingouin/parametric.py#L516

which should actually be:

```python
data = data.groupby([subject, within, between], observed=True).mean().reset_index()
```

I'm guessing they are other functions that are similarly affected so I'll need some time to update and test everything, but I'll definitely include this in the next stable version of Pingouin (sometimes early September).

Thanks,
Raphael

**Comment by raphaelvallat** [comment]
This is fixed in the just-released stable version of Pingouin (v0.3.8). Please make sure to update your version. Closing the issue. Thanks again!
